C/plugins/storage/sqlitememory/connection.cpp

```diff
@@ -111,3 +113,78 @@ bool Connection::vacuum()
 {
 	return true;
 }
+
+/**
+ * Load the in memory database from a file backup
+ *
+ * @param filename	The name of the file to restore from
+ * @return bool		Success or failure of the backup
+ */
+bool Connection::loadDatabase(const string& filename)
+{
+int rc;
+sqlite3 *file;
+sqlite3_backup *backup;
+
+	string pathname = getDataDir() + "/";
+	pathname.append(filename);
+	pathname.append(".db");
+	if (access(pathname.c_str(), R_OK) != 0)
+	{
+		Logger::getLogger()->warn("Persisted database %s does not exist",
+				pathname.c_str());
+		return false;
+	}
+	if ((rc = sqlite3_open(pathname.c_str(), &file)) == SQLITE_OK)
+	{
+		if (backup = sqlite3_backup_init(dbHandle, READINGS_TABLE_MEM, file, "main"))
+		{
+			(void)sqlite3_backup_step(backup, -1);
+			(void)sqlite3_backup_finish(backup);
+			Logger::getLogger()->info("Reloaded persisted data to in memory database");
+		}
+		rc = sqlite3_errcode(dbHandle);
+
+		(void)sqlite3_close(file);
+	}
+	return rc == SQLITE_OK;
+}
+
+/**
+ * Backup the in memory database to a file
+ *
+ * @param filename	The name of the file to backup to
+ * @return bool		Success or failure of the backup
+ */
+bool Connection::saveDatabase(const string& filename)
+{
+int rc;
+sqlite3 *file;
+sqlite3_backup *backup;
+
+	string pathname = getDataDir() + "/";
+	pathname.append(filename);
+	pathname.append(".db");
+	unlink(pathname.c_str());
+	if ((rc = sqlite3_open(pathname.c_str(), &file)) == SQLITE_OK)
+	{
+		if (backup = sqlite3_backup_init(file, "main", dbHandle, READINGS_TABLE_MEM))
+		{
+			rc = sqlite3_backup_step(backup, -1);
+			(void)sqlite3_backup_finish(backup);
+			Logger::getLogger()->info("Persisted data from in memory database to %s", pathname.c_str());
```

Comment:
Perhaps be consistent with the dash in "in-memory?" This message is without; the messages below have the dash. This occurs in several places in the file.